### PR TITLE
Page Resource Block Rules Avoid Duplicate Handlers + Ignore top-level pages + README update (0.4.4)

### PR DESCRIPTION
**Pull request by ikreymer**
Avoid adding duplicate interception handlers on same page object, which caused errors indicated in #80

Update README to clarify difference between page resource block rules (blockRules) and seed/crawl scope rules.

Update docs to indicate param renamed to `scopeType` instead of `type`.

Block Rules do not apply to top-level pages (causes page load errors), instead warning printed to use scope rules if intent is to exclude full pages.

Also: update `page` scope to really be a single page (was `none`) before while page + hashtags scope renamed to `page-spa`

Ensure current working directory exists by creating it in case of a custom `--cwd` points to a custom path.

bump version to 0.4.4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -218,38 +218,84 @@ Certain settings such scope type, scope includes and excludes, and depth can als
 seeds:
   - url: https://webrecorder.net/
     depth: 1
-    type: "prefix"
+    scopeType: "prefix"
```
Comment (emmadickson) [inline README.md:221]: I like this rename

File: util/seeds.js [hunk 2]
```diff
@@ -78,10 +82,6 @@ class ScopedSeed
       include = [/.*/];
       break;
```
Comment (emmadickson) [inline util/seeds.js:84]: should we be removing the none case?

**Review by emmadickson — APPROVED** [review]
Looks good to me. The readme update is really clear

File: util/seeds.js (hunk 2 above)
Comment (ikreymer) [reply util/seeds.js:84]: yeah, wasn't very clear before, now updated in README and the error message to print the valid values for scopeType.

**Pull request merged**